commit 7084a3993fa35c6fb71abe8aac7b30f442205e2a
Author: Lev Stipakov <lstipakov@gmail.com>
Date:   Wed Dec 7 01:45:51 2016 +0200

    Arm inotify only in server mode
    
    Async-push is a server side feature and inotify_fd is
    initialized in server mode.
    
    Trac: #786
    Signed-off-by: Lev Stipakov <lstipakov@gmail.com>
    Acked-by: David Sommerseth <davids@openvpn.net>
    Message-Id: <1481067951-28917-1-git-send-email-lstipakov@gmail.com>
    URL: https://www.mail-archive.com/openvpn-devel@lists.sourceforge.net/msg13415.html
    Signed-off-by: David Sommerseth <davids@openvpn.net>

diff --git a/src/openvpn/forward.c b/src/openvpn/forward.c
index b50a2e0..4502e10 100644
--- a/src/openvpn/forward.c
+++ b/src/openvpn/forward.c
@@ -1578,7 +1578,8 @@ io_wait_dowork (struct context *c, const unsigned int flags)
 
 #ifdef ENABLE_ASYNC_PUSH
   /* arm inotify watcher */
-  event_ctl (c->c2.event_set, c->c2.inotify_fd, EVENT_READ, (void*)&file_shift);
+  if (c->options.mode == MODE_SERVER)
+    event_ctl (c->c2.event_set, c->c2.inotify_fd, EVENT_READ, (void*)&file_shift);
 #endif
 
   /*
